File: lintcode/133_longest-words/longest-words.py
# coding:utf-8
'''
@Copyright:LintCode
@Author:   chenweikang
@Problem:  http://www.lintcode.com/problem/longest-words
@Language: Python
@Datetime: 16-04-25 01:11
'''

class Solution:
    # @param dictionary: a list of strings
    # @return: a list of strings
    def longestWords(self, dictionary):
        longest = 0
        res = []
        for word in dictionary:
            longest = max(longest, len(word))
        for word in dictionary:
            if len(word) == longest:
                res.append(word)
        return res

File: lintcode/133_longest-words/longest-words.cpp
/*
@Copyright:LintCode
@Author:   chenweikang
@Problem:  http://www.lintcode.com/problem/longest-words
@Language: C++
@Datetime: 16-05-12 00:34
*/

class Solution {
public:
    /**
     * @param dictionary: a vector of strings
     * @return: a vector of strings
     */
    vector<string> longestWords(vector<string> &dictionary) {
        vector<string> res;
        int longest = 0;
        for (auto &word : dictionary) {
            if (word.size() > longest) {
                longest = word.size();
                res.clear();
                res.push_back(word);
            } else if (word.size() == longest) {
                res.push_back(word);
            }
        }
        return res;
    }
};